### ISSUE TITLE
Detached window that receives new event should get urgency hint, rather than main qTox window (blinking)

### ISSUE THREAD

**Issue by zetok** [qTox/qTox#2211]
Description: When detached window receives new event, main qTox window gets urgency hint (blinking). Instead, detached window that received urgency hint should get urgency hint.

OS: Lubuntu (15.04?)

Reproducible: Always(?)

Additional info:
![fubqfsl](https://cloud.githubusercontent.com/assets/3148759/9833399/1e50fb18-598d-11e5-8690-471442efe309.png)
↑ `not having blue` window should have an urgency hint, not the main window

First reported by @ovalseven8 in https://github.com/tux3/qTox/pull/1938#issuecomment-134951140

**Comment by zetok** [comment]
From #2316, thanks to @ByteCommander:

# 

I have set **Multiple windows mode** in `qtox-unity` on Ubuntu 15.04, so that the left main window (contact list) and chat/settings windows are separated and independent of each other.

Now when have the main window and a chat window open, both are not focussed and in background, the tray icon starts blinking when I receive a message (of the user in my open chat window). But when I now click on the chat window so that it gets focussed, the blinking tray icon does not stop! Same if the main window is closed and only the chat window is open in background.

The blinking only stops when I (open, if necessary, and ) focus the main window.

**Proposed attempt to fix:**
Don't switch the tray blinking off when the main window gets focussed, but when the respective chat window gets focussed.

---

From #2316, thanks to @FabrizioMarana:

# 

Same here.  

lsb_release --short --codename&&uname --kernel-release 
trusty
3.13.0-63-generic

Commit hash: 42c29c8b8fa5e719c9c78c15f109b358c9225c0c

Any other info you need: shout!

**Comment by ByteCommander** [comment]
Yes, same issue indeed. I just did not recognize it when searching through the titles...

**Comment by ovalseven8** [comment]
'P-medium' to my mind.

**Comment by Chiitoo** [comment]
It seems this issue may have transformed a little bit due to certain changes to the alerts.

If the main window is minimised, alerts are triggered for chat windows even if they are the active window.  If the main window is not minimised, however, things seem to work as they should.

Also, opening the main window then clears the blinking icon, but not the 'New message' from the window title of the chat.

Also also, the taskbar entry does not start flashing when the active window gets the alert, even though it does get the 'New message', and the icon starts blinking in the system tray.

**Edit:**

Oh and if the chat window is not the active window, or one switches to another window and then back, it will clear the blinking of the tray icon, too, since the mentioned changes (I think).
